### PR TITLE
Rhb 48

### PR DESCRIPTION
**Pull request by Avaxapsys**
I'm going to update tests during PR checking

### CODE REVIEW

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs [hunk 1]
```diff
@@ -28,40 +34,50 @@ public class CourseController : Controller
...
+        public CourseInfoShortOutputModel GetCourseSimple(int id)
         {
             var course = _courseService.GetCourse(id);
-            return _mapper.Map<CourseInfoFullOutputModel>(course);
+            return _mapper.Map<CourseInfoShortOutputModel>(course);
         }
 
         [HttpGet("{id}/full")]
         [Description("Get course by id full")]
+        [AuthorizeRoles(Role.Admin, Role.Manager, Role.Teacher, Role.Methodist)]
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/CourseController.cs:53]: Admin must be excluded from this list
please check the implementation of AuthorizeRolesAttribute and you will find the reason

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs [hunk 2]
```diff
@@ -28,40 +34,50 @@ public class CourseController : Controller
...
-            return _mapper.Map<List<CourseInfoFullOutputModel>>(courses);
+            return _mapper.Map<List<CourseInfoShortOutputModel>>(courses);
         }
 
 
         [HttpPost]
         [Description("Create new course")]
-        [ProducesResponseType(typeof(CourseInfoFullOutputModel), StatusCodes.Status201Created)]
-        public CourseInfoFullOutputModel AddCourse([FromBody] CourseInputModel model)
+        [AuthorizeRoles(Role.Admin, Role.Manager, Role.Teacher, Role.Methodist)]
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/CourseController.cs:77]: and here as well
in addition, Teacher can't create a course

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs [hunk 3]
```diff
@@ -70,16 +86,20 @@ public CourseInfoFullOutputModel AddCourse([FromBody] CourseInputModel model)
 
         [HttpDelete("{id}")]
         [Description("Delete course by id")]
+        [Authorize]
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/CourseController.cs:89]: really?)

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs [hunk 4]
```diff
@@ -70,16 +86,20 @@ public CourseInfoFullOutputModel AddCourse([FromBody] CourseInputModel model)
...
         public void DeleteCourse(int id)
         {
             _courseService.DeleteCourse(id);
         }
 
         [HttpPut("{id}")]
         [Description("Update course by Id")]
-        [ProducesResponseType(typeof(CourseInfoFullOutputModel), StatusCodes.Status200OK)]
-        public CourseInfoFullOutputModel UpdateCourse(int id, [FromBody] CourseInputModel model)
+        [Authorize(Roles = "Admin, Manager")]
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/CourseController.cs:99]: wrong attribute

File: DevEdu.API/DevEdu.API/Controllers/GroupController.cs [hunk 5]
```diff
@@ -88,7 +90,10 @@ public GroupOutputBaseModel ChangeGroupStatus(int groupId, int statusId)
         // api/Group/{groupId}/lesson/{lessonId}
         [HttpPost("{groupId}/lesson/{lessonId}")]
         [Description("Add group lesson reference")]
+        [Authorize(Roles = "Admin, Manager, Teacher")]
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/GroupController.cs:93]: wrong attribute

File: DevEdu.API/DevEdu.API/Controllers/GroupController.cs [hunk 6]
```diff
@@ -98,7 +103,10 @@ public string AddGroupToLesson(int groupId, int lessonId)
         // api/Group/{groupId}/lesson/{lessonId}
         [HttpDelete("{groupId}/lesson/{lessonId}")]
         [Description("Delete lesson from groupId")]
+        [Authorize(Roles = "Admin, Manager, Teacher")]
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/GroupController.cs:106]: wrong attribute

File: DevEdu.API/DevEdu.Business/Services/CourseService.cs [hunk 7]
```diff
@@ -15,42 +16,61 @@ public class CourseService : ICourseService
...
+            _courseValidation.CheckCourseExistence(id);
+            _courseRepository.DeleteCourse(id);
+        }
 
-        public CourseDto GetCourse(int id) => _courseRepository.GetCourse(id);
-        public CourseDto GetFullCourseInfo(int id) 
+        public CourseDto GetCourse(int id)
+        {
+            var course = _courseRepository.GetCourse(id);
+            _courseValidation.CheckCourseExistence(course.Id);
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business/Services/CourseService.cs:48]: you are getting the same course twice

File: DevEdu.API/DevEdu.Business/ValidationHelpers/CourseValidationHelper.cs [hunk 8]
```diff
@@ -13,11 +14,20 @@ public CourseValidationHelper(ICourseRepository courseRepository)
             _courseRepository = courseRepository;
         }
 
-        public void CheckCourseExistence(int courseId)
+        public CourseDto CheckCourseExistence(int courseId)
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business/ValidationHelpers/CourseValidationHelper.cs:17]: please rename to GetCourseByIdAndThrowIfNotFound

File: DevEdu.API/DevEdu.DAL/Repositories/BaseRepository.cs [hunk 9]
```diff
@@ -6,7 +6,8 @@ namespace DevEdu.DAL.Repositories
     public abstract class BaseRepository
     {
         protected const string _connectionString =
-            @"Data Source=80.78.240.16;Initial Catalog = DevEdu; Persist Security Info=True;User ID = student;Password=qwe!23;";
+            //@"Data Source=80.78.240.16;Initial Catalog = DevEdu; Persist Security Info=True;User ID = student;Password=qwe!23;";
+        @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog = DevEdu;Trusted_Connection=true";
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/BaseRepository.cs:10]: please don't commit this)

File: DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs [hunk 10]
```diff
@@ -316,5 +317,15 @@ public int GetPresentGroupForStudentByUserId(int userId)
                    commandType: CommandType.StoredProcedure
               );
         }
+
+        public List<GroupDto> GetGroupByCourseId(int courseId)
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs:321]: please fix naming

File: DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs [hunk 11]
```diff
@@ -316,5 +317,15 @@ public int GetPresentGroupForStudentByUserId(int userId)
                    commandType: CommandType.StoredProcedure
               );
         }
+
+        public List<GroupDto> GetGroupByCourseId(int courseId)
+        {
+            var groupList = new List<GroupDto>();
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs:323]: you don't need this

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs (hunk 1 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.API/Controllers/CourseController.cs:53]: +

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs (hunk 2 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.API/Controllers/CourseController.cs:77]: +

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs (hunk 3 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.API/Controllers/CourseController.cs:89]: +

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs (hunk 4 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.API/Controllers/CourseController.cs:99]: +

File: DevEdu.API/DevEdu.API/Controllers/GroupController.cs (hunk 5 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.API/Controllers/GroupController.cs:93]: +

File: DevEdu.API/DevEdu.Business/Services/CourseService.cs (hunk 7 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business/Services/CourseService.cs:48]: +

File: DevEdu.API/DevEdu.Business/ValidationHelpers/CourseValidationHelper.cs (hunk 8 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business/ValidationHelpers/CourseValidationHelper.cs:17]: +

File: DevEdu.API/DevEdu.DAL/Repositories/BaseRepository.cs (hunk 9 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.DAL/Repositories/BaseRepository.cs:10]: +

File: DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs (hunk 10 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs:321]: +

File: DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs (hunk 11 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs:323]: +

**Pull request merged**